src/validation.cpp

```diff
@@ -769,6 +770,11 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
             std::vector<EthTransactionParams> qtumETP = resultConverter.second;
             for(QtumTransaction qtumTransaction : qtumTransactions){
                 sumGas += CAmount(qtumTransaction.gas() * qtumTransaction.gasPrice());
+
+                if(qtumTransaction.gas() > blockGasLimit){
+                    return state.DoS(100, false, REJECT_INVALID, "bad-txns-gas-exceeds-blockgaslimit");
```

Comment:
This should be an Invalid error rather than DoS error. Users may mistakenly send a tx with a lot of gas at the same time a DGP proposal causes the block gas to be reduced